homework/App.js

```diff
@@ -14,16 +14,37 @@ class App {
   async initialize(url) {
     // Add code here to initialize your app
     // 1. Create the fixed HTML elements of your page
-    // 2. Make an initial XMLHttpRequest using Util.fetchJSON() to populate your <select> element
 
     const root = document.getElementById('root');
-
-    Util.createAndAppend('h1', root, { text: 'It works!' }); // TODO: replace with your own code
+    const header = Util.createAndAppend('header', root, { class: 'header' });
+    Util.createAndAppend('h1', header, {
+      class: 'h1',
+      text: 'HYF Repositories',
+    });
+    const select = Util.createAndAppend('select', header);
+    Util.createAndAppend('div', root, { class: 'repo-div', id: 'containers' });
+    select.addEventListener('change', event => {
+      this.fetchContributorsAndRender(event.target.value);
+    });
 
     try {
-      const repos = await Util.fetchJSON(url);
-      this.repos = repos.map(repo => new Repository(repo));
-      // TODO: add your own code here
+      const data = await fetch(url);
+      const fetchedData = await (function error() {
+        if (data.ok) {
+          return data.json();
+        }
+        throw new Error('There is fetching error');
+      })();
+      this.repos = fetchedData
+        .sort((a, b) => a.name.localeCompare(b.name))
+        .map(repo => new Repository(repo));
+      fetchedData.forEach((repo, index) => {
+        Util.createAndAppend('option', select, {
+          text: repo.name,
+          value: index,
+        });
+      });
+      this.fetchContributorsAndRender([0]);
```

Comment:
This method expects a number as an argument. You are passing it an array with a single numeric element. That the code works all the same is pure coincidence.

Reply:
i  just fixed it 
thanks a lot!